addons/gamification/models/res_users.py

```diff
@@ -42,35 +64,46 @@ def _get_user_badge_level(self):
     @api.model_create_multi
     def create(self, values_list):
         res = super(Users, self).create(values_list)
-
-        karma_trackings = []
-        for user in res:
-            if user.karma:
-                karma_trackings.append({'user_id': user.id, 'old_value': 0, 'new_value': user.karma})
+        karma_trackings = [
+            {
+                'new_value': vals.get('karma'),
+                'old_value': 0,
+                'origin_ref': f'res.users,{self.env.uid}',
+                'reason': _('User Creation'),
+                'user_id': user.id,
+            }
+            for user, vals in zip(res, values_list)
+            if vals.get('karma')
+        ]
         if karma_trackings:
             self.env['gamification.karma.tracking'].sudo().create(karma_trackings)
 
-        res._recompute_rank()
         return res
 
-    def write(self, vals):
```

Comment:
Why do we remove the support of "write" ?

Would be nice, as people are probably going to write on this field (whether we want it or not, for example by adding the field in the form view).

Reply:
If it's not very complicated to do I would support it, I'm almost certain people are going to do it, isn't it "just" calling _add_karma_batch with appropriate values after super?